cmake/OpenCVDumpPreprocessor.cmake

```diff
@@ -0,0 +1,35 @@
+# Utility
+function(dump_gcc_preprocessor fname exe extra_flags)
+  set(src "${CMAKE_BINARY_DIR}/preprocessor/empty")
+  file(WRITE "${src}" "")
+  string(REPLACE " " ";" opt "${extra_flags}")
+  execute_process(COMMAND "${exe}" ${opt} -dM -E -
+    RESULT_VARIABLE res
+    OUTPUT_VARIABLE out
+    ERROR_VARIABLE ERR
+    INPUT_FILE "${src}"
+  )
+  string(REPLACE "\n" ";" out "${out}")
```

Comment:
`res` is not verified.
`ERR` is not used.

It makes sense to dump extra information to the output file:
- `"${exe}" ${opt}` - to see used options
- `${res}`
- `${ERR}`

Reply:
Done:
```
 head -n15 dump_CXX_RELEASE_baseline.txt 
CMD => 0
/usr/bin/c++    -fsigned-char -W -Wall -Wreturn-type -Wnon-virtual-dtor -Waddress -Wsequence-point -Wformat -Wformat-security -Wmissing-declarations -Wundef -Winit-self -Wpointer-arith -Wshadow -Wsign-promo -Wuninitialized -Wsuggest-override -Wno-delete-non-virtual-dtor -Wno-comment -Wimplicit-fallthrough=3 -Wno-strict-overflow -fdiagnostics-show-option -Wno-long-long -pthread -fomit-frame-pointer -ffunction-sections -fdata-sections  -msse -msse2 -msse3 -fvisibility=hidden -fvisibility-inlines-hidden -O3 -DNDEBUG  -DNDEBUG

STDERR
cc1: warning: command-line option ‘-Wnon-virtual-dtor’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-Wsign-promo’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-Wsuggest-override’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-Wno-delete-non-virtual-dtor’ is valid for C++/ObjC++ but not for C
cc1: warning: command-line option ‘-fvisibility-inlines-hidden’ is valid for C++/ObjC++ but not for C

===

#define NDEBUG 1
#define _FORTIFY_SOURCE 2
#define _LP64 1
```

Reply:
> `cc1: warning: command-line option ‘-Wnon-virtual-dtor’ is valid for C++/ObjC++ but not for C`

to avoid these warnings add `-x c++` through `CMAKE_CXX_COMPILE_OPTIONS_EXPLICIT_LANGUAGE` variable